lib/reverse_words.rb

```diff
@@ -1,6 +1,52 @@
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of the string. The code loops through every index of the string.
+# Space complexity: O(1) because no matter the length of the string, the following variables will constantly take up space:
+# my_words, first_index, last_index, string_last, i, j, temp
+
 def reverse_words(my_words)
-  raise NotImplementedError
+  if my_words == nil
+    return nil
+  end
+  first_index = 0
+  last_index = 0
+  string_last = my_words.length - 1
+  my_words.each_char do |char|
```

Comment:
A do loop with a break is a valid way to do things, but it can be hard to read. Instead, consider a while loop with a condition like (char != " ")